connector/src/yang/connector/grpc.py

```diff
@@ -0,0 +1,179 @@
+                        config.write(f)
+                else:
+                    # set config file path
+                    self.config_file = f"{runtime.directory}/telegraf.conf"
+                    config.read(self.config_file)
+
+                    # if the file already exists, only update the port
+                    if config.sections():
+                        if '[inputs.cisco_telemetry_mdt]' not in config.sections():
+                            config.add_section('[inputs.cisco_telemetry_mdt]')
+
+                        config.set('[inputs.cisco_telemetry_mdt]', 'transport', '"grpc"')
+                        config.set('[inputs.cisco_telemetry_mdt]', 'service_address', f'":{allocated_port}"')
+
+                        with open(self.config_file, 'w') as f:
+                            log.info(f"Writing config to {self.config_file}")
+                            config.write(f)
+                    else:
+                        # generate a default configuration file
+                        with open(self.config_file, 'w') as f:
+                            log.info(f"Creating telegraf config file {self.config_file}")
+                            f.write('')
+
+                        # create default config
+                        # global tags
+                        config.add_section('global_tags')
+                        config.set('global_tags', 'user', r'"${USER}"')
+
+                        # input configuration
+                        config.add_section('[inputs.cisco_telemetry_mdt]')
+                        config.set('[inputs.cisco_telemetry_mdt]', 'transport', '"grpc"')
+                        config.set('[inputs.cisco_telemetry_mdt]', 'service_address', f'":{allocated_port}"')
+
+                        # default output config - to file in runtime or user supplied dir
+                        config.add_section('[outputs.file]')
+                        config.set('[outputs.file]', 'files', f'["stdout", "{self.output_file}"]')
+                        config.set('[outputs.file]', 'data_format', '"json"')
+                        config.set('[outputs.file]', 'json_timestamp_units', '"1ms"')
+                        config.set('[outputs.file]', 'rotation_max_size', '"2048MB"')
+                        config.set('[outputs.file]', 'flush_jitter', '"500ms"')
+
+                        # apply config
+                        with open(self.config_file, 'w') as f:
+                            log.info(f"Updating {self.config_file}")
+                            config.write(f)
+
+        # exit context manager to release port
+        # spawn telegraf/pipeline using config
+        self.telegraf_process = subprocess.Popen(f"telegraf -config '{self.config_file}'", shell=True)
+
+        # log port
+        log.info(f"Telegraf is running as PID {self.telegraf_process.pid} on port {allocated_port}")
+
+        # call the API to genie configure the service on the device
+        self.device.connect()
```

Comment:
Please instantiate the connection first and check if its a unicon connection. If its not a unicon connection, raise a failure.